### ISSUE TITLE
`sourceMapString` is not a function [v5.28.0]

### ISSUE THREAD

**Issue by devphilipcesar** [getsentry/sentry-react-native#4005]
**OS:**
- [ ]  Windows
- [X]  MacOS
- [ ]  Linux

**Platform:**
- [X]  iOS
- [X]  Android

**SDK:**
- [X]  `@sentry/react-native` (>= 1.0.0)
- [ ]  `react-native-sentry` (<= 0.43.2)

**SDK version:** 5.26.0

**`react-native` version:** 0.73.9

**Are you using Expo?**
- [x]  Yes
- [ ]  No

**Are you using sentry.io or on-premise?**
- [X]  sentry.io (SaaS)
- [ ]  on-premise

**If you are using sentry.io, please post a link to your issue so we can take a look:**

[Link to issue]

**Configuration:**

(`@sentry/react-native`)
```
Sentry.init({
  dsn: 'https://...@sentry.io/...'
  // other options
});
```

or

(`react-native-sentry`)
```
Sentry.config(
  'https://...@sentry.io/...'
  // other options
 ).install();
```

---
**I have the following issue:**

[Description]

**Steps to reproduce:**
- Step 1
- Step 2

**Actual result:**

![Image](https://github.com/user-attachments/assets/a3e920c2-1f80-43e2-a268-aaf11614f6db)

**ANDROID**

![Image](https://github.com/user-attachments/assets/0904ae38-a09f-4cb4-a332-a1a4850812e3)

**iOS**

![Image](https://github.com/user-attachments/assets/bc45478b-abe3-43de-bc3f-bf8538b50348)

**Expected result:**

[Expected result]

**Comment by gretzky** [comment]
Seeing the same issue.

Expo SDK 51
`@sentry/react-native` v5.22.0

![Image](https://github.com/user-attachments/assets/38565793-ff82-4023-8de9-cb13e29fcc0b)

**Comment by devphilipcesar** [comment]
[UPDATE]

I was able to fix issue by removing the `createSentryMetroSerializer` and the whole `serializer`


**Old file:**

![Image](https://github.com/user-attachments/assets/bf91ad55-564d-4552-b227-cb278f7a8bcb)



**Updated file:**



![Image](https://github.com/user-attachments/assets/89780c1f-78b0-48a0-bf5e-4c8cad0d6c6b)

**Comment by matthijsFoF** [comment]
We're having the same issue, is removing this createSentryMetroSerializer really the way to go here? 

This was also generated by the sentry wizard.

```
TypeError: sourceMapString is not a function
error sourceMapString is not a function.
```

**Comment by devphilipcesar** [comment]
Hi @matthijsFoF  basically at that time I bump the sentry react native version to 5.27.0 and removed that `createSentryMetroSerializer`... I tested using the Sentry test error and I was able to track the error on Sentry Dashboard with no issues on sourcemaps.

UPDATE:
update to 5.28.0 will resolve the issue.

**Comment by gretzky** [comment]
We aren't using `createSentryMetroSerializer` in our config, so I'm not sure that resolves the issue fully. We are also on 5.28.0 and still seeing the error, @devphilipcesar do you mind reopening this issue? 

Logs:


![Image](https://github.com/user-attachments/assets/4283ad8a-eb9f-4b43-b17d-b1f6e2232800)



Metro config file:

```js
const { getSentryExpoConfig } = require("@sentry/react-native/metro");
const { FileStore } = require("metro-cache");
const path = require("path");

module.exports = withTurborepoManagedCache(
  withMonorepoPaths(getSentryExpoConfig(__dirname))
);

function withMonorepoPaths(config) {
  const projectRoot = __dirname;
  const workspaceRoot = path.resolve(projectRoot, "../..");

  config.watchFolders = [workspaceRoot];

  config.resolver.nodeModulesPaths = [
    path.resolve(projectRoot, "node_modules"),
    path.resolve(workspaceRoot, "node_modules"),
  ];

  return config;
}

function withTurborepoManagedCache(config) {
  config.cacheStores = [
    new FileStore({ root: path.join(__dirname, "node_modules/.cache/metro") }),
  ];
  return config;
}
```

**Comment by matthijsFoF** [comment]
Hi @devphilipcesar, fixed for us now, thanks for the quick response!

**Comment by atitpatel** [comment]
Our appcenter builds have started failing due to the same error.
React native: 0.74.1
Different sentry packages:
```
"@sentry/cli": "2.32.2",
"@sentry/integrations": "7.101.1",
"@sentry/react-native": "5.20.0",
```
These builds were working fine till Friday last week and then started failing without any library change from our end. Any help is appreciated here. Thanks!

**Comment by krystofwoldrich** [comment]
Hello everyone,
The issue was caused by the latest Metro update changing the `sourceMapString` function export.

It was fixed by this [PR](https://github.com/getsentry/sentry-react-native/pull/4004).

- https://github.com/getsentry/sentry-react-native/pull/4004

Please upgrade to https://github.com/getsentry/sentry-react-native/releases/tag/5.28.0

**Comment by krystofwoldrich** [comment]
> I tried upgrading @sentry/react-native to 5.28.0, but kept getting build errors around Sentry/HybridSDK and various attempts to update pods didn't help it.

@atitpatel Could you share what errors are you getting?

**Comment by draftbitbuildservices** [comment]
Hi, I'm facing the same issues, already updated to 5.28.0
```

❌  error: sentry-cli - To disable source maps auto upload, set SENTRY_DISABLE_AUTO_UPLOAD=true in your environment variables. Or to allow failing upload, set SENTRY_ALLOW_FAILURE=true



❌  error: sentry-cli -   INFO    2024-08-16 17:10:00.874586 +00:00 Loaded file referenced by SENTRY_PROPERTIES (sentry.properties)



❌  error: (0 , sourceMapString_1.default) is not a function

```

**Comment by sieu-db** [comment]
Hi @krystofwoldrich 
Seem like 

```
/Users/builder/clone/node_modules/@sentry/cli-darwin/bin/sentry-cli /Users/builder/clone/node_modules/@expo/cli/build/bin/cli export:embed --entry-file node_modules/expo/AppEntry.js --platform ios --dev false --reset-cache --bundle-output /Users/builder/Library/Developer/Xcode/DerivedData/TridentMindset-epkqrcgtafzjrlddyzgubdhysvqa/Build/Intermediates.noindex/ArchiveIntermediates/TridentMindset/BuildProductsPath/Release-iphoneos/main.jsbundle --assets-dest /Users/builder/Library/Developer/Xcode/DerivedData/TridentMindset-epkqrcgtafzjrlddyzgubdhysvqa/Build/Intermediates.noindex/ArchiveIntermediates/TridentMindset/BuildProductsPath/Release-iphoneos/TridentMindset.app --sourcemap-output /Users/builder/Library/Developer/Xcode/DerivedData/TridentMindset-epkqrcgtafzjrlddyzgubdhysvqa/Build/Intermediates.noindex/ArchiveIntermediates/TridentMindset/BuildProductsPath/Release-iphoneos/main.jsbundle.map --minify false
error: (0 , sourceMapString_1.default) is not a function
TypeError: (0 , sourceMapString_1.default) is not a function
TypeError: (0 , sourceMapString_1.default) is not a function
    at /Users/builder/clone/node_modules/expo/node_modules/@expo/metro-config/build/serializer/withExpoSerializers.js:98:55
    at /Users/builder/clone/node_modules/expo/node_modules/@expo/metro-config/build/serializer/withExpoSerializers.js:156:20
    at Object.customSerializer (/Users/builder/clone/node_modules/expo/node_modules/@expo/metro-config/build/serializer/withExpoSerializers.js:182:16)
    at Server.build (/Users/builder/clone/node_modules/metro/src/Server.js:156:52)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async exportEmbedBundleAsync (/Users/builder/clone/node_modules/@expo/cli/build/src/export/embed/exportEmbedAsync.js:134:16)
    at async exportEmbedBundleAndAssetsAsync (/Users/builder/clone/node_modules/@expo/cli/build/src/export/embed/exportEmbedAsync.js:122:24)
    at async exportEmbedAsync (/Users/builder/clone/node_modules/@expo/cli/build/src/export/embed/exportEmbedAsync.js:62:34)
```

**Comment by sieu-db** [comment]
Hi, we are also on 5.28.0 and still seeing the error, @devphilipcesar @krystofwoldrich do you mind reopening this issue ?

Expo 50
Sentry 5.28.0

The same code build without issues ~2 weeks ago, but right now facing error below

```
error: (0 , sourceMapString_1.default) is not a function
TypeError: (0 , sourceMapString_1.default) is not a function
TypeError: (0 , sourceMapString_1.default) is not a function
    at /Users/builder/clone/node_modules/expo/node_modules/@expo/metro-config/build/serializer/withExpoSerializers.js:98:55
    at /Users/builder/clone/node_modules/expo/node_modules/@expo/metro-config/build/serializer/withExpoSerializers.js:156:20
    at Object.customSerializer (/Users/builder/clone/node_modules/expo/node_modules/@expo/metro-config/build/serializer/withExpoSerializers.js:182:16)
    at Server.build (/Users/builder/clone/node_modules/metro/src/Server.js:156:52)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async exportEmbedBundleAsync (/Users/builder/clone/node_modules/@expo/cli/build/src/export/embed/exportEmbedAsync.js:134:16)
    at async exportEmbedBundleAndAssetsAsync (/Users/builder/clone/node_modules/@expo/cli/build/src/export/embed/exportEmbedAsync.js:122:24)
    at async exportEmbedAsync (/Users/builder/clone/node_modules/@expo/cli/build/src/export/embed/exportEmbedAsync.js:62:34)
+ '[' -z '' ']'
++ /usr/local/bin/node --print 'require('\''path'\'').dirname(require.resolve('\''@sentry/react-native/package.json'\''))'
```

**Comment by Njinou** [comment]
@sieu-db p lease any update on that issue ?

**Comment by jeremyfrancis** [comment]
**TLDR;
npx expo install --fix
rm -rf yarn.lock package-lock.json node_modules
yarn**

Think about a node module you recently installed, I found this is usually due to a node module expecting a newer version of react native.  Running "npx expo install --fix" has resolved this for me as I had a package from @react-native-community that was too new of a version.  It somehow knew to downgrade it and then it worked.

So really I need to update my RN at some point.

After this correction, I recommend removing package-lock.json or yarn.lock and removing node_modules to reinstall.

**Comment by krystofwoldrich** [comment]
If you are an Expo user, ensure `getSentryExpoConfig` is executed.

If you are a Bare RN user, use `withSentryConfig`.

https://docs.sentry.io/platforms/react-native/manual-setup/metro/#use-the-sentry-metro-plugin

If you still see this error, please open a new issue with details about your setup, RN, Expo, and Metro versions....
